app/src/main/java/co/nimblehq/compose/crypto/ui/components/linechart/CoinPriceChart.kt

```diff
@@ -0,0 +1,143 @@
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.drawscope.drawIntoCanvas
+import androidx.compose.ui.unit.dp
+import co.nimblehq.compose.crypto.ui.components.chartintervals.ChartIntervalsButtonGroup
+import co.nimblehq.compose.crypto.ui.components.chartintervals.TimeIntervals
+import me.bytebeats.views.charts.line.*
+import me.bytebeats.views.charts.line.render.line.EmptyLineShader
+import me.bytebeats.views.charts.line.render.line.ILineDrawer
+import me.bytebeats.views.charts.line.render.line.ILineShader
+import me.bytebeats.views.charts.line.render.line.SolidLineDrawer
+import me.bytebeats.views.charts.line.render.point.FilledCircularPointDrawer
+import me.bytebeats.views.charts.line.render.point.IPointDrawer
+import me.bytebeats.views.charts.line.render.xaxis.IXAxisDrawer
+import me.bytebeats.views.charts.line.render.xaxis.SimpleXAxisDrawer
+import me.bytebeats.views.charts.line.render.yaxis.IYAxisDrawer
+import me.bytebeats.views.charts.line.render.yaxis.SimpleYAxisDrawer
+import me.bytebeats.views.charts.simpleChartAnimation
+
+@Suppress("MagicNumber", "LongMethod")
+@Composable
+fun CoinPriceChart(
+    lineChartData: LineChartData,
+    modifier: Modifier = Modifier,
+    animation: AnimationSpec<Float> = simpleChartAnimation(),
+    pointDrawer: IPointDrawer = FilledCircularPointDrawer(),
+    lineDrawer: ILineDrawer = SolidLineDrawer(),
+    lineShader: ILineShader = EmptyLineShader,
+    xAxisDrawer: IXAxisDrawer = SimpleXAxisDrawer(),
+    yAxisDrawer: IYAxisDrawer = SimpleYAxisDrawer(),
+    labelDrawer: ILabelDrawer = CoinPriceLabelDrawer(),
+    horizontalOffset: Float = 5F,
+    onTimeIntervalChanged: (TimeIntervals) -> Unit
+) {
+    check(horizontalOffset in 0F..25F) {
+        "Horizontal Offset is the percentage offset from side, and must be between 0 and 25, included."
+    }
+    val transitionAnimation = remember(lineChartData.points) { Animatable(initialValue = 0F) }
+
+    LaunchedEffect(lineChartData.points) {
+        transitionAnimation.snapTo(0F)
+        transitionAnimation.animateTo(1F, animationSpec = animation)
+    }
+
+    Column(
+        modifier = modifier.fillMaxWidth(),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Canvas(modifier = modifier.height(184.dp)) {
+            drawIntoCanvas { canvas ->
+                val yAxisDrawableArea = computeYAxisDrawableArea(
+                    xAxisLabelSize = xAxisDrawer.requireHeight(this),
+                    size = size
+                )
+                val xAxisDrawableArea = computeXAxisDrawableArea(
+                    yAxisWidth = yAxisDrawableArea.width,
+                    labelHeight = xAxisDrawer.requireHeight(this),
```

Comment:
Actually, we do nothing at `requireHeight`, should we create a DEFAULT_require_Height = 0 instead? We could get rid of `xAxisDrawer` and `yAxisDrawer` params, and then don't need to define `EmptyXAxisDrawer` and `EmptyYAxisDrawer`?

Reply:
```suggestion
                    xAxisLabelSize = DEFAULT_REQUIRE_HEIGHT,
                    size = size
                )
                val xAxisDrawableArea = computeXAxisDrawableArea(
                    yAxisWidth = yAxisDrawableArea.width,
                    labelHeight = DEFAULT_REQUIRE_HEIGHT,
```